packages/expo-cli/src/commands/upload/IOSUploader.ts

```diff
@@ -110,16 +121,19 @@ export default class IOSUploader extends BaseUploader {
     };
   }
 
-  async _getAppleTeamId() {
+  async _getAppleTeamId(): Promise<string | undefined> {
     const credentialMetadata = await Credentials.getCredentialMetadataAsync(this.projectDir, 'ios');
-    const { teamId } = await Credentials.getCredentialsForPlatform(credentialMetadata);
+    const credential = await Credentials.getCredentialsForPlatform(credentialMetadata);
 
-    return teamId;
+    if (credential) {
+      return credential.teamId;
+    }
+    return;
   }
 
-  async _getAppleIdCredentials() {
+  async _getAppleIdCredentials(): Promise<IosPlatformOptions> {
```

Comment:
This function does not return a full `IosPlatformOptions` object, it seems, only the subset `{  appleId: string;  appleIdPassword: string }` of it.
```suggestion
  async _getAppleIdCredentials(): Promise<Pick<IosPlatformOptions, 'appleId', 'appleIdPassword'>> {
```